lib/kubebox.js

```diff
@@ -153,6 +153,86 @@ class Kubebox extends EventEmitter {
       carousel.move();
     });
 
+    dashboard.on('exec', (namespace, pod, container) => {
+      // check if connection already exists
+      const user = kube_config.current_context.user.username;
+      const id = namespace + pod + container;
+      function wsexecExists (_function) {
+        if (!_function.id)
+          return false;
+        return _function.id === id;
+      }
+      const index = carousel.pages.findIndex((wsexecExists));
+      if (index >= 0) {
+        carousel.currPage = index;
+        carousel.move();
+        return;
+      }
+      
+      // connect
+      const exec = new Exec(screen, namespace, pod, container, user);
+      const { promise, cancellation, sendData, getSocket} = get(client.exec(namespace, pod, container, exec.termName()), { stream: () => exec.print() });
+      promise
+        .then(() => debug.log(`{grey-fg}Remote shell in container: '${container}', pod: '${pod}' and namespace: '${namespace}' with user '${user}'{/grey-fg}...`))
+        .then(() => {
+          exec.sendData = sendData;
+          const exit = () => {
+            const index = carousel.pages.findIndex((wsexecExists));
+            if (index >= 0)
+              carousel.pages.splice(index, 1);
+            cancellation();
+            exec.kill();
+            tabs.removeItem(index);
```

Comment:
In the same spirit, it may be better to create a `menu` widget that encapsulates the tabs logic so that it can be reused elsewhere.

Reply:
Let me work on this.

Reply:
Done in 3a6800b653a9bc1cedbed1bf1e11a1c7490ab0d9.